pageserver/src/tenant/tasks.rs

```diff
@@ -103,6 +99,7 @@ async fn compaction_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
                 }
             }
 
+            // FIXME: stopping will come before cancellation, so unsure if this will help often
```

Comment:
@jcsp do you think it's worth while to keep this check? As far as I think, we already have the cancelled check with the wait on top of this loop, and below on the wait for some time, so for me it's extra.

Reply:
This check can probably go away.

My change from earlier in the week in theory catches the narrow race where we exited `random_init_delay` without cancellation, but were cancelled before entering `compaction_iteration`.  But it would be better to just put an `if` around the error log line, to suppress it in all conditions where we reach an error state but are already shutting down: that way there's no risk that any earlier cancellation checks race with the tenant state check in compaction_iteration.

Reply:
On the other PR I also noted something about tenant entering stopping *before* these tasks are cancelled.

I'll remove it.

Reply:
[fd507f0](https://github.com/neondatabase/neon/pull/4886/commits/fd507f0be51e7c1bc1e2c4fde1cda688f2e3b5f3)